### PR TITLE
Replace x11 CDT deps with xorg- packages

### PR DESCRIPTION
**Pull request by ryanvolz**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ryanvolz** [comment]
I tried for a little while to get this compiling on osx-arm64 with the latest meson, but I haven't been able to make it work yet. So that'll have to wait for the future, for the sake of finishing the xorg package transition.

File: recipe/post-link.bat [hunk 1]
```diff
@@ -1,6 +1,19 @@
-"%PREFIX%\Library\bin\glib-compile-schemas.exe" "%PREFIX%\Library\share\glib-2.0\schemas"
-if errorlevel 1 exit 1
-"%PREFIX%\Library\bin\gtk-update-icon-cache.exe" -f -t "%PREFIX%\Library\share\icons\hicolor"
-if errorlevel 1 exit 1
-"%PREFIX%\Library\bin\gtk-query-immodules-3.0.exe" > "%PREFIX%\Library\lib\gtk-3.0\3.0.0\immodules.cache"
-if errorlevel 1 exit 1
+:: When cross-compiling, or installing for a different platform, the commands
```
Comment (tschoonj) [inline recipe/post-link.bat:1]: Is cross compiling with Windows currently done on conda-forge?

File: conda-forge.yml [hunk 2]
```diff
@@ -1,3 +1,4 @@
+azure: {store_build_artifacts: true}
```
Comment (tschoonj) [inline conda-forge.yml:1]: I assume this is a leftover from testing?

File: recipe/bld.bat [hunk 3]
```diff
@@ -24,15 +24,22 @@ set ^"MESON_OPTIONS=^
...
-ninja -v -C builddir -j %CPU_COUNT%
+ninja -v -C forgebuild -j %CPU_COUNT%
 if errorlevel 1 exit 1
 
-ninja -C builddir install -j %CPU_COUNT%
+ninja -C forgebuild install -j %CPU_COUNT%
+if errorlevel 1 exit 1
+
+:: create directory for modules so post-link script doesn't fail
+set "MODULEDIR=%LIBRARY_LIB%\gtk-3.0\3.0.0"
```
Comment (tschoonj) [inline recipe/bld.bat:41]: Why is this necessary now? I thought this folder was created during `ninja install`?

File: recipe/post-link.bat (hunk 1 above)
Comment (ryanvolz) [reply recipe/post-link.bat:1]: Cross-compiling? No. I read the comment as describing some but not necessarily all situations where this is useful.

File: conda-forge.yml (hunk 2 above)
Comment (ryanvolz) [reply conda-forge.yml:1]: Leftover in the sense that it was added while I was testing, but I find it generally useful to have in place all the time so you don't have to stick it in and re-run the CI to analyze the output.

File: recipe/bld.bat (hunk 3 above)
Comment (ryanvolz) [reply recipe/bld.bat:41]: Well, when I was trying out the error catching, I found that this has been silently erroring the whole time. That directory is not created during the install because the modules are built statically on Windows (a default upstream choice). So while nothing was broken with the built in modules because of this, it was still erroring. That error apparently wasn't caught because it was in the shell redirection and not in the program itself, so the errorlevel was not set. With this in there, the (empty) directory is now a part of the package as the tools would expect

**Comment by pkgw** [comment]
The implementation here is looking good.

For background, I generally heard two reasons for using X11 via CDT rather than the xorg- packages. These may be discussed in more detail in my ancient (but still open!) PR conda-forge/cfep#7.

- One reason was to maintain compatibility with the Anaconda recipes.
- The other reason was concerns about interfaces with system libraries farther down the stack, namely the OpenGL layer. The worry was that maybe custom conda-forge X11 libraries would be stuck using software GL instead of the best, potentially vendor-specific, library for the host machine.

This discussion was happening all before the CDT concept even was defined. I don't know if there are well-defined GL CDTs, but if so, that might obviate the second issue. As for the first issue, I feel like I haven't heard the general issue of recipe compatibility mentioned recently in any context ... I don't have a good sense for exactly how the topic has shaken out. It's worth chasing down for an important platform library, but there's also only so much we can do without external input.

**Comment by ryanvolz** [comment]
>     * The other reason was concerns about interfaces with system libraries farther down the stack, namely the OpenGL layer. The worry was that maybe custom conda-forge X11 libraries would be stuck using software GL instead of the best, potentially vendor-specific, library for the host machine.
> 
> 
> This discussion was happening all before the CDT concept even was defined. I don't know if there are well-defined GL CDTs, but if so, that might obviate the second issue.

I do think that if there were an issue here, it would have already come up. There are GL CDTs, and it is (still) what `epoxy` uses to connect the GTK stack to OpenGL. From the other side, this package on conda-forge has effectively been running with the `xorg-libx11` dependency and friends the whole time, because they are always installed in the environment by `cairo`. So even though we have been building with CDTs, we've been running with conda-forge's `xorg`. And so far nobody has complained (or maybe even noticed). Anyway, as far as I understand it, users have always and will continue to be able to run the stack with hardware GL support.

Still, broader discussion around these topics would be good so we don't catch conda-forge core unawares. I guess I'll go ping https://github.com/conda-forge/cfep/pull/7, since our own bug https://github.com/conda-forge/gtk3-feedstock/issues/7 doesn't have nearly as much visibility.

**Comment by pkgw** [comment]
Thanks for working through this. My overall feeling is similar: I'm not aware of any actual complaints about GL issues or the like, so I'm inclined to say it's fine. But the worry is worth taking seriously, and it's also good to go the extra mile to work well with Anaconda.com. Unfortunately I haven't had the bandwidth to pursue the topic for a long time (i.e. multiple years) but I'd really love to converge on a solid policy for how to approach X11 — I believe that it's the most significant difference between the conda-forge and `defaults` frameworks.

**Comment by pkgw** [comment]
@conda-forge/core Any comments on this proposal? I'm generally in favor but am not up to speed on where things are at with the broader policy questions.

**Comment by hmaarrfk** [comment]
I just had this discussion with @isuruf during a core meeting. 

He feels strongly that the only concern we should have with the usage of CDTs is when they hamper a critical feature (like the usage of OpenGL).

I feel that we should strive for compatibility with Anaconda.org, but I do not think it is worthwhile at this moment to sacrifice the usage of in-built x11 packages.

Ultimately, this difference is "pretty easy to spot" in the sense that I don't think it requires any changes within the `build.sh` file itself. The build system typically find X11 from either the CDT, or the host dependencies with little intervention (in my experience).

xref: https://github.com/conda-forge/qt-main-feedstock/pull/114

**Comment by hmaarrfk** [comment]
I also meant to reference the meeting minutes https://github.com/conda-forge/conda-forge.github.io/blob/main/src/orga/minutes/2023-02-08.md
